src/escpos/config.py

```diff
@@ -45,6 +45,33 @@ def _reset_config(self) -> None:
         self._printer_name = None
         self._printer_config = None
 
+    def _set_config(self, config) -> None:
+        """Set configuration variables.
+
+        :param config: config object loaded from file or string.
+        """
+        if "printer" in config:
+            self._printer_config = config["printer"]
+            printer_name = self._printer_config.pop("type")
+            class_names = {
+                "usb": "Usb",
+                "serial": "Serial",
+                "network": "Network",
+                "file": "File",
+                "dummy": "Dummy",
+                "cupsprinter": "CupsPrinter",
+                "lp": "LP",
+                "win32raw": "Win32Raw",
+            }
+            self._printer_name = class_names.get(printer_name.lower(), printer_name)
+
+            if not self._printer_name or not hasattr(printer, self._printer_name):
+                raise exceptions.ConfigSyntaxError(
+                    f'Printer type "{self._printer_name}" is invalid'
+                )
```

Comment:
Mypy complains about this block: 'src/escpos/config.py:55: error: "None" has no attribute "pop"'
This would fix it:

```suggestion
        self._printer_config = config.get("printer")
        if not self._printer_config:
            return

        printer_name = self._printer_config.pop("type")
        class_names = {
            "usb": "Usb",
            "serial": "Serial",
            "network": "Network",
            "file": "File",
            "dummy": "Dummy",
            "cupsprinter": "CupsPrinter",
            "lp": "LP",
            "win32raw": "Win32Raw",
        }
        self._printer_name = class_names.get(printer_name.lower(), printer_name)

        if not self._printer_name or not hasattr(printer, self._printer_name):
            raise exceptions.ConfigSyntaxError(
                f'Printer type "{self._printer_name}" is invalid'
            )
```

Reply:
Hi @belono. Thanks for having a look. I have a small nit to pick with this one: `load` currently sets  `_has_loaded` to True even when a printer is not found:

```
        if "printer" in config:
            self._printer_config = config["printer"]
         #...   
         
        self._has_loaded = True
```


We should set `_has_loaded` to True before the early return to maintain the same behavior:

```
    if not self._printer_config:
        self._has_loaded = True
        return
```

Reply:
Also if we use a generic falsy check here (`if not self._printer_config:`), then a config with an empty dict like `printer: {}` would be treated as "no printer section" and accepted.

In the original implementation this case actually raised a KeyError when trying to pop "type"

Reply:
I agree. The falsy check `if not self._printer_config:` is incorrect. We should check for `None`.

> then a config with an empty dict like printer: {} would be treated as "no printer section" and accepted

We really should add a test for this case. Something like this approach:
```python
from escpos import config

c = config.Config()
with pytest.raises(escpos.exceptions.ConfigSyntaxError):
    c.load_yaml_string("printer:\n   {}")
```

Reply:
Currently, we'd need to check for the KeyError

```
    with pytest.raises(KeyError):
        c.load_yaml_string("printer: {}")
```
Unless you want to catch it and re-raise it as ConfigSyntaxError

```
if self._printer_config is None:
    self._has_loaded = True
    return
try:
    printer_name = self._printer_config.pop("type")
except KeyError:
    raise exceptions.ConfigSyntaxError('...')
```

Reply:
> We should set _has_loaded to True before the early return to maintain the same behavior:

After a second look at the old and the new code I realized that there is another way not involving early return and maintaining a similar behavior.

I'll made a code suggestion.